src/dali/plugin/pair_converter/csv/kraken.py

```diff
@@ -141,19 +157,23 @@ class Kraken:
 
     __DELIMITER: str = ","
 
-    def __init__(
-        self,
-        google_api_key: str,
-    ) -> None:
-        self.__google_api_key: str = google_api_key
+    def __init__(self, transaction_manifest: TransactionManifest, force_download: bool = False, unified_csv_file_id: str = _UNIFIED_CSV_FILE_ID) -> None:
```

Comment:
Do we need the unified_csv_file parameter: since this plugin is meant to download the hardcoded file pointed to in _UNIFIED_CSV_FILE_ID, is there ever a legitimate use for passing a different value? I'm concerned about potential security problems if we leave this flexibility, so I wanted to know if we would lose any desirable functionality by removing this parameter.

Reply:
Dali RP2 will need to get updated with a new file id every 3 months when a new unified csv file is released. This was meant to let them choose a new file id until we get an update out.

But, they can just delete all the CSVs and download the new file themselves, so it isn't really necessary. Should I just remove it?

Reply:
Yes, I think it's better to remove the parameter: since this plugin is downloading a file from the Web, it's better to keep the reference to the file hardcoded rather than parametric to avoid possible misuse. Like you said, the workaround would be to download the file manually until we update the plugin.